shift_detector/precalculations/FrequentItemsetPrecalculation.py

```diff
@@ -0,0 +1,27 @@
+from shift_detector.Utils import ColumnType
+from shift_detector.precalculations.frequent_item_rules import fpgrowth
+from shift_detector.precalculations.frequent_item_rules import rule_compression
+from shift_detector.precalculations.Precalculation import Precalculation
+
+
+class FrequentItemsetPrecalculation(Precalculation):
+
+    def __init__(self, min_support=0.01, min_confidence=0.15):
+        self.min_support = min_support
+        self.min_confidence = min_confidence
+
+    def process(self, store):
+        df1, df2 = store[ColumnType.categorical]
+
+        item_rules = fpgrowth.calculate_frequent_rules(df1, df2, self.min_support, self.min_confidence)
+        compressed_rules = rule_compression.compress_rules(item_rules)
+
+        return {'compressed_rules': compressed_rules, 'examined_columns': set(df1.columns)}
```

Comment:
Wouldn't it be simpler to just use 'rules' as the key? Or is the differentiation between compressed and uncompressed relevant to the outer scope somehow? As only the compressed rules are returned I'd assume not.

Reply:
because the compressed rules are very different from the initially created rules